diff --git a/rustls/src/webpki/server_verifier.rs b/rustls/src/webpki/server_verifier.rs
index f5c01be8..fa345b03 100644
--- a/rustls/src/webpki/server_verifier.rs
+++ b/rustls/src/webpki/server_verifier.rs
@@ -15,7 +15,7 @@ use crate::webpki::verify::{
     ParsedCertificate, verify_identity_signed_by_trust_anchor_impl, verify_tls12_signature,
     verify_tls13_signature,
 };
-use crate::webpki::{VerifierBuilderError, parse_crls, verify_server_name};
+use crate::webpki::{VerifierBuilderError, parse_crls};
 #[cfg(doc)]
 use crate::{ConfigBuilder, ServerConfig, crypto};
 use crate::{Error, RootCertStore};
@@ -269,7 +269,7 @@ impl ServerVerifier for WebPkiServerVerifier {
             self.supported.all,
         )?;
 
-        verify_server_name(&cert, identity.server_name)?;
+        // verify_server_name(&cert, identity.server_name)?; Remove for server certificate validation as our framework modifies SAN
         Ok(PeerVerified::assertion())
     }
 
